src/org/mozilla/javascript/NativeJSON.java

```diff
@@ -91,11 +91,11 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
             case Id_stringify: {
                 Object value = null, replacer = null, space = null;
                 switch (args.length) {
+                    default:
```

Comment:
I'm a bit conflicted about this since it's a very clever switch statement, but it's also a bit confusing. Would a series of "if (args.length >= 1)", "if (args.length >= 2)", and such statements be easier to follow for future maintainers? I realize that I'm suggesting three comparisons versus (theoretically) one, but it does seem easier to follow. I don't have a strong opinion, though.

Reply:
I mentioned in the issue that I opened that before the regression bug was introduced by moving the default case to the bottom, the original switch statement was written with the default case at the top. I think it looked a lot cleaner than the cascading if statements. Perhaps it only needs a comment that the default case is for >= 3?
 ```
              switch (args.length) {
                    /* >= 3     */ default: space = args[2];
                    /* fallthru */ case 2: replacer = args[1];
                    /* fallthru */ case 1: value = args[0];
                    /* fallthru */ case 0:
                }  
```